includes/class-wc-stripe-helper.php

```diff
@@ -1727,6 +1727,11 @@ public static function get_klarna_preferred_locale( $store_locale, $billing_coun
 	 */
 	public static function add_mandate_data( $request ) {
 		$ip_address = WC_Geolocation::get_ip_address();
+
+		// Handle cases where WC_Geolocation::get_ip_address() returns multiple, comma-separated IP addresses.
+		// TODO: Remove this block when WooCommerce 9.9.0 is released.
+		$ip_address = trim( current( preg_split( '/,/', sanitize_text_field( wp_unslash( $ip_address ) ) ) ) );
```

Comment:
I have two thoughts on this approach:
1. We should link to the upstream PR in the comments so we can easily pick up the thread and context later
2. Given that we're not expecting this to happen, it feels like we're going to do a lot of work here for every call. Is it worth adding a simple gate that checks for the `,` character before triggering all the cleanup code?
3. I don't think we need to unslash or sanitize the value in our code, as it should have been sanitized already in `get_ip_address()`. AFAICT, we're cleaning up the values in that function, so we shouldn't need to re-process the values in this code.
```suggestion
		// Handle cases where WC_Geolocation::get_ip_address() returns multiple, comma-separated IP addresses.
		// This will be addressed upstream in WooCommerce 9.9.0 as of (https://github.com/woocommerce/woocommerce/pull/57284).
		// TODO: Remove this block when WooCommerce 9.9.0 is released.
		if ( str_contains( $ip_address, ',' ) ) {
			$ip_address = trim( current( preg_split( '/,/', $ip_address ) ) );
		}
```

Reply:
@daledupreez Good call, committed your suggestion.